chromium_src/components/update_client/update_checker.cc

```diff
@@ -0,0 +1,102 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "../../../../components/update_client/update_checker.cc"
+
+namespace update_client {
+
+SequentialUpdateChecker::SequentialUpdateChecker(
+    scoped_refptr<Configurator> config,
+    PersistedData* metadata)
+    : config_(config), metadata_(metadata) {
+      VLOG(3) << "SequentialUpdateChecker";
+    }
+
+SequentialUpdateChecker::~SequentialUpdateChecker() {
+    VLOG(3) << "> ~SequentialUpdateChecker";
+    DCHECK(thread_checker_.CalledOnValidThread());
+    VLOG(3) << "< ~SequentialUpdateChecker";
+}
+
+void SequentialUpdateChecker::CheckForUpdates(
+    const std::string& session_id,
+    const std::vector<std::string>& ids_checked,
+    const IdToComponentPtrMap& components,
+    const base::flat_map<std::string, std::string>& additional_attributes,
+    bool enabled_component_updates,
+    UpdateCheckCallback update_check_callback) {
+  VLOG(3) << "> CheckForUpdates";
+
+  DCHECK(thread_checker_.CalledOnValidThread());
+  DCHECK(!ids_checked.empty());
+
+  for (const auto& app_id : ids_checked) {
+    VLOG(3) << ">  * " << app_id;
+    remaining_ids_.push_back(app_id);
+  }
+
+  session_id_ = session_id;
+  components_ = &components;
+  additional_attributes_ = additional_attributes;
+  enabled_component_updates_ = enabled_component_updates;
+  update_check_callback_ = std::move(update_check_callback);
+
+  CheckNext();
+
+  VLOG(3) << "< CheckForUpdates";
+}
+
+void SequentialUpdateChecker::CheckNext() {
+  VLOG(3) << "> CheckNext()";
+  DCHECK(!remaining_ids_.empty());
+  std::string id = remaining_ids_.front();
+  remaining_ids_.pop_front();
+  std::vector<std::string> id_vector = {id};
+
+  update_checker_ = UpdateChecker::Create(config_, metadata_);
+  update_checker_->CheckForUpdates(
```

Comment:
Looks to me like this will result in a new thread (in `UpdateCheckerImpl::CheckForUpdates`) for each component in the ids list, is that a concern?

Reply:
Or maybe I am wrong here since we only check for the next id when the result of the previous one becomes available.